### PR TITLE
Connector builder projects: List and get APIs

### PR DESCRIPTION
**Pull request by flash1293**
## What

Based on https://github.com/airbytehq/airbyte-platform/pull/45

Fixes https://github.com/airbytehq/airbyte/issues/22445
Fixes https://github.com/airbytehq/airbyte/issues/22573

By adding APIs to list all builder projects of a workspace as well as fetching a single project with its manifest.

## How

* Extend open api spec: `airbyte-api/src/main/openapi/config.yaml`
* Extend the builder project fetching with a calculated field `hasDraft` (`airbyte-config/config-models/src/main/resources/types/ConnectorBuilderProject.yaml`, `airbyte-config/config-persistence/src/main/java/io/airbyte/config/persistence/DbConverter.java`, `airbyte-config/config-persistence/src/main/java/io/airbyte/config/persistence/ConfigRepository.java`) which is used for the listing to know whether there is a draft without actually fetching it.
* Fix an existing problem of not storing a string `"null"` instead of actually no value if there is no manifest draft defined (also adding tests for this) - this happened because JSON serialization will turn a Java `null` into a JSON string `"null"`. Added a branch to https://github.com/airbytehq/airbyte/pull/23143/files#diff-2d29b293778e290157be96e22294c2df1fdfa7c72a48961d03396776446deeddR1663 to fix this

For testing and general notes see description of https://github.com/airbytehq/airbyte/pull/23063

## Can this PR be safely reverted / rolled back?
*If you know that your PR is backwards-compatible and can be simply reverted or rolled back, check the YES box.*

*Otherwise if your PR has a breaking change, like a database migration for example, check the NO box.*

*If unsure, leave it blank.*
- [x] YES 💚
- [ ] NO ❌

## 🚨 User Impact 🚨

No impact

### CODE REVIEW

**Comment by flash1293** [comment]
Port from `airbyte` repo https://github.com/airbytehq/airbyte/pull/23143

cc @maxi297 @jdpgrailsdev @girarda @xiaohansong

File: airbyte-api/src/main/openapi/config.yaml [hunk 1]
```diff
@@ -557,6 +557,44 @@ paths:
       responses:
         "204":
           description: Successful operation
+  /v1/connector_builder_projects/list:
+    post:
+      tags:
+        - connector_builder_project
+      summary: List connector builder projects for workspace
+      operationId: listProjects
```
Comment (xiaohansong) [inline airbyte-api/src/main/openapi/config.yaml:565]: a naming nit: can this be listConnectorBuilderProjects to avoid confusion? Project sounds a pretty big word here and could cover many other things

File: airbyte-config/config-persistence/src/main/java/io/airbyte/config/persistence/ConfigRepository.java [hunk 2]
```diff
@@ -1627,8 +1631,7 @@ public Stream<ConnectorBuilderProject> getConnectorBuilderProjectsByWorkspace(fi
 
     return database
         .query(ctx -> ctx
-            .select(CONNECTOR_BUILDER_PROJECT.ID, CONNECTOR_BUILDER_PROJECT.WORKSPACE_ID, CONNECTOR_BUILDER_PROJECT.NAME,
-                CONNECTOR_BUILDER_PROJECT.ACTOR_DEFINITION_ID)
+            .select(baseConnectorBuilderProjectColumns)
             .from(CONNECTOR_BUILDER_PROJECT)
             .where(matchByIds)
```
Comment (xiaohansong) [inline airbyte-config/config-persistence/src/main/java/io/airbyte/config/persistence/ConfigRepository.java:1636]: We probably want to sort on something in this query (like create time or just name) to ensure the order will stay consistent

File: airbyte-api/src/main/openapi/config.yaml (hunk 1 above)
Comment (flash1293) [reply airbyte-api/src/main/openapi/config.yaml:565]: The create/delete/update operation ids are named the same way - I will adjust across the board.

**Comment by flash1293** [comment]
Thanks for the review @xiaohansong , adjusted naming and introduced explicit sorting (also reflected in the tests)

**Comment by flash1293** [comment]
@lmossman Check out the change I made - this is how the types look like now:

When reading a builder project, it's this type:
```
export interface ConnectorBuilderProjectDetailsRead {
  name: string;
  builderProjectId: ConnectorBuilderProjectId;
  hasDraft: boolean;
}
```

Used in the listing response like this:

```
export interface ConnectorBuilderProjectReadList {
  projects: ConnectorBuilderProjectDetailsRead[];
}
```

When writing a builder project, it's this type:
```
export interface ConnectorBuilderProjectDetails {
  name: string;
  draftManifest?: ConnectorBuilderProjectDetailsDraftManifest;
}
```

Used like this in the create request:
```
export interface ConnectorBuilderProjectWithWorkspaceId {
  workspaceId: WorkspaceId;
  builderProject: ConnectorBuilderProjectDetails;
}
```

and like this in the update request:
```
export interface ExistingConnectorBuilderProjectWithWorkspaceId {
  workspaceId: WorkspaceId;
  builderProjectId: ConnectorBuilderProjectId;
  builderProject: ConnectorBuilderProjectDetails;
}
```

Also, the problem with `hasDraft` never going to `false` should be fixed on this PR.

**Review by xiaohansong — APPROVED** [review]

**Comment by lmossman** [comment]
@flash1293 those changes make sense to me, thanks for the summary!

**Review by girarda — APPROVED** [review]
:shipit:
